site/guides/sponsoring-gas/gas-sponsorship-eligibility.md

```diff
@@ -0,0 +1,119 @@
+
+```ts [check-gas-sponsorship-eligibility.ts]
+let elligible = true;
+
+try {
+  await provider.buildUserOperation({
+    target: "0xTargetAddress",
+    data: "0xCallData",
+    value: 0n, // value in bigint or leave undefined
+  });
+} catch (e) {
+  // if the user operation is not eligible for gas sponsorship, error will be thrown and caught here
+  elligible = false;
+}
+
+console.log(
+  `User Operation is ${
+    elligible ? "elligible" : "inelligible"
+  } for gas sponsorship`
+);
+```
+
+## 2. How to Bypass the Paymaster Middleware For User Operations Not Eligible for Gas Sponsorship
+
+So how would you handle the user operation not elligible for gas sponsorship? You might want to unblock your users to be able to make the user operation transaction without gas sponsorship. This section of the guide explains how you can bypass the `PaymasterMiddleware` and still be able to send the user operation without gas sponsorship. This will let your users to make the transaction by paying the gas fee themselves.
+
+When you call `sendUserOperation` method on `SmartAccountProvider`, in addition to the `UserOperationCallData` or `BatchUserOperationCallData`, you can pass in an optional param `overrides` where you can specify override values for `maxFeePerGas`, `maxPriorityFeePerGas` or `paymasterAndData`. In order to bypass the `PaymasterMiddleware`, you can specify the `paymasterAndData` override. For example, if you set the `paymasterAndData` field in the overrides as `0x`, this will make the user operation to be sent where the gas fee is not paid by the paymaster but by the user's own smart account.
+
+```ts [bypass-paymaster-middleware.ts]
+// Baypass paymaster middleware by passing in the paymasterAndData override
+
+/*
+ * export type UserOperationOverrides = Partial<
+ *   Pick<
+ *     UserOperationStruct,
+ *     "maxFeePerGas" | "maxPriorityFeePerGas" | "paymasterAndData"
+ *   >
+ * >;
+ */
+
+// Empty paymasterAndData indicates that there will be no paymaster involved
+// and the user will be paying for the gas fee even when `withAlchemyGasManager` is configured on the provider
+const overrides = { paymasterAndData: "0x" };
+
+const userOperationResult = await provider.sendUserOperation(
+  {
+    target: "0xTargetAddress",
+    data: "0xCallData",
+    value: 0n,
+  },
+  overrides
+);
+
+const txHash = await provider.waitForUserOperationTransaction({
+  hash: userOperationResult.hash,
+});
+```
+
+That's it! By using the above methods to:
+
+1. Check for the gas sponsorship eligibility prior to sending the user operation
+2. Bypass the paymster middleware for user operations not elligible for your gas manager policy
+
+AccountKit provides much flexibility for you to design optimal user experiences to handle different cases of gas sponsonship eligibility accordingly.
```

Comment:
This doc should further clarify that the dapp does 1) and then acts on it in code with 2). Specifically, the user clicks button to send the UO, which the dapp should check if eligible. If not eligible, the dapp should inform the user and ask if they would want to kick off 2) (i.e. pay for gas themselves). 

Right now reading this, it's a bit unclear how dev should incorporate 1+ 2 in the app flow.

See my comment above re: `tip` section on advising good UX.